sisyphus/simple_linux_utility_for_resource_management_engine.py

```diff
@@ -393,28 +396,66 @@ def get_default_rqmt(self, task):
 
     def init_worker(self, task):
         # setup log file by linking to engine logfile
-        task_id = self.get_task_id(None)
-        logpath = os.path.relpath(task.path(gs.JOB_LOG, task_id))
+
+        # Naming ambiguity: sis "tasks" are what SLURM calls array jobs.
+        #
+        # SLURM tasks represent jobs that span multiple nodes at the same time
+        # (e.g. multi-node multi-GPU trainings consist of one SLURM task per node).
+        slurm_num_tasks = int(
+            next(filter(None, (os.getenv(var, None) for var in ["SLURM_NTASKS", "SLURM_NPROCS"])), "1")
+        )
+        slurm_task_id = int(os.getenv("SLURM_PROCID", "0"))
+
+        array_task_id = self.get_task_id(None)
+        # keep backwards compatibility: only change output file name for multi-SLURM-task jobs
+        log_suffix = array_task_id if slurm_num_tasks <= 1 else f"{array_task_id}.{slurm_task_id}"
+        logpath = os.path.relpath(task.path(gs.JOB_LOG, log_suffix))
         if os.path.isfile(logpath):
             os.unlink(logpath)
 
-        engine_logpath = (
-            os.path.dirname(logpath)
-            + "/engine/"
-            + os.getenv("SLURM_JOB_NAME")
-            + "."
-            + os.getenv("SLURM_ARRAY_JOB_ID")
-            + "."
-            + os.getenv("SLURM_ARRAY_TASK_ID")
+        job_id = next(
+            filter(None, (os.getenv(name, None) for name in ["SLURM_JOB_ID", "SLURM_JOBID", "SLURM_ARRAY_JOB_ID"])), "0"
         )
-        try:
-            if os.path.isfile(engine_logpath):
+        has_linked_logfile = False
+        engine_logpath_candidates = [
+            (
+                os.path.dirname(logpath)
+                + "/engine/"
+                + os.getenv("SLURM_JOB_NAME")
+                + "."
+                + job_id
+                + "."
+                + str(slurm_task_id)
+                + "."
+                + os.getenv("SLURM_ARRAY_TASK_ID", "1")
+            ),
+            (
+                os.path.dirname(logpath)
+                + "/engine/"
+                + os.getenv("SLURM_JOB_NAME")
+                + "."
+                + job_id
+                + "."
+                + os.getenv("SLURM_ARRAY_TASK_ID", "1")
+            ),
+        ]
+        for engine_logpath in engine_logpath_candidates:
```

Comment:
I don't think we need this.. we do already know if this is a  multi-SLURM-task job. So just select the correct engine_logpath.